### ISSUE TITLE
[bug] conan client can't handle "provides" attribute as a collection

### ISSUE THREAD

**Issue by SpaceIm** [conan-io/conan#7621]
### Environment Details (include every applicable attribute)
  * Operating System+version: all
  * Compiler+version: all
  * Conan version: 1.28.1
  * Python version: 3.7.1

### Steps to reproduce (Include if Applicable)

In `conanfile.py` of a recipe, add `provides` attribute with at least two recipes listed (https://github.com/conan-io/conan-center-index/pull/2716/commits/b62f5188fb5026e9c27e468fc9511a01d9c480be for example), then run `conan create`

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
ERROR: module 'six.moves' has no attribute 'collections_abc'
```

**Comment by kralicky** [comment]
Related: #7520

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/7545 implemented a fix for Conan 1.29, but lets backport to next Conan 1.28.2 too.

**Comment by czoido** [comment]
Closed by https://github.com/conan-io/conan/pull/7622
Fix will be released very soon in Conan 1.28.2
